eth-bridge/contracts/src/WrappedToken.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import {ERC20} from "openzeppelin-contracts/contracts/token/ERC20/ERC20.sol";
+import {Pausable} from "openzeppelin-contracts/contracts/security/Pausable.sol";
+import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import {ERC20Permit} from "openzeppelin-contracts/contracts/token/ERC20/extensions/draft-ERC20Permit.sol";
+
+contract WrappedToken is ERC20, Pausable, Ownable, ERC20Permit {
```

Comment:
I dont understand this file tbh

Reply:
This implements the ERC-20 token that represents our LLM or LLD bridged to Ethereum. It's based on the OpenZeppelin's ERC-20 implementation: https://docs.openzeppelin.com/contracts/4.x/erc20 .

The `contract WrappedToken is [...]` is basically an inheritance, and this is what we get from each of those:
* `ERC20` - tracking balances, supporting transfers between accounts
* `Pausable` - being able to pause transfers
* `Ownable` - gives the contract an Owner - we use it to for authorization of some function calls - see all the `onlyOwner`.
* `ERC20Permit` - signature-based approvals - basically cheaper transactions if clients support it

The only custom thing in this file is the infinite allowance for contract owner (a.k.a. the bridge). Everything else is boilerplate OpenZeppelin configuration stuff.